packages/app/cypress/e2e/runner/cloud-debug-filter.cy.ts

```diff
@@ -0,0 +1,85 @@
+describe('cloud debug test filtering', () => {
+  it('works with nested suites', () => {
+    cy.scaffoldProject('cloud-debug-filter')
+    cy.openProject('cloud-debug-filter')
+    cy.startAppServer('e2e')
+    cy.visitApp(`specs/runner?file=cypress/e2e/test.cy.js`)
+
+    cy.waitForSpecToFinish()
+
+    cy.withCtx((ctx) => {
+      ctx.coreData.cloud.testsForRunResults = [{ titlePath: 't2', status: 'FAILED' }]
+    })
+
+    cy.visitApp(`specs/runner?file=cypress/e2e/test.cy.js&runId=123`)
+    cy.waitForSpecToFinish({ passCount: 0, failCount: 1 })
+
+    cy.get('.runnable-title').contains('t2')
+
+    cy.get('.debug-dismiss').contains('1 / 4 tests').click()
+    cy.waitForSpecToFinish({ passCount: 2, failCount: 2 })
+
+    cy.withCtx((ctx) => {
+      ctx.coreData.cloud.testsForRunResults = [{ titlePath: 's1 t4', status: 'FAILED' }]
+    })
+
+    cy.visitApp(`specs/runner?file=cypress/e2e/test.cy.js&runId=123`)
+    cy.waitForSpecToFinish({ passCount: 0, failCount: 1 })
+
+    cy.get('.runnable-title').contains('t4')
+  })
+
+  it('works with skips and onlys', () => {
```

Comment:
can we add a test case for browser skip?

Reply:
Added in [5e431a0](https://github.com/cypress-io/cypress/pull/25265/commits/5e431a05532e2116e048ad3468ede9aa41ff6eda)!